### DISCUSSION TITLE
Using avahi-daemon: no such host

### DISCUSSION THREAD

**Discussion by nidkil** [smallstep/certificates#594] [General] [answered]
Hi,

I have avahi-daemon running that publishes the servers on my internal network. When I ping the server running Step CA it works. When I try and bootstrap a server using the server name (tiny-ca.internal) I get `dial tcp: lookup tiny-ca.internal on 127.0.0.53:53: no such host` error. Any idea's why I can ping the Step CA server, but the step command doesn't work?

The Step CA command I'm executing.

```
sudo -u ubuntu step ca bootstrap --ca-url="https://tiny-ca.internal" --fingerprint 9bb37fcff164bdf5bc993a95c3362755c899eee291f8a4321bbf681e3dec4cfa 
```

The error message.

```
error downloading root certificate: client.Root; client GET https://tiny-ca.internal/root/9bb37fcff164bdf5bc993a95c3362755c899eee291f8a4321bbf681e3dec4cfa failed: Get "https://tiny-ca.internal/root/9bb37fcff164bdf5bc993a95c3362755c899eee291f8a4321bbf681e3dec4cfa": dial tcp: lookup tiny-ca.internal on 127.0.0.53:53: no such host
```

The command does work when I use the servers IP address.

**Comment by tashian** [comment, marked as answer]
Just wanted to note two possible solutions to this, for posterity:
1. Build step-ca [with CGO enabled](https://github.com/smallstep/certificates/blob/master/docs/CONTRIBUTING.md#build-step-ca-using-cgo).
2. Or, if you're in a systemd-based environment, systemd-resolved now has a built-in multicast DNS resolver. As an alternative to using avahi-daemon, turn on the mDNS resolver in `/etc/systemd/resolved.conf`. See https://wiki.archlinux.org/title/Systemd-resolved#mDNS
